### ISSUE TITLE
Strings consisting '\t' or '\v' not interpreted in YAML

### ISSUE THREAD

**Issue by Lord-KA** [tarantool/tarantool#7681]
When using escaped tab in string, YAML stops interpreting all escaped chars.  

* OS: Linux
* OS Version: Ubuntu 22
* Architecture: x86_64

master

**Steps to reproduce**
```
Tarantool 2.10.0-beta1-1165-g0b622ac1a
type 'help' for interactive help
tarantool> a = '\t\n\n'
---
...

tarantool> a
---
- "\t\n\n"
...

tarantool> b = '\v\n\n'
---
...

tarantool> b
---
- "\v\n\n"
...

tarantool> c = '\\\n\n'
---
...

tarantool> c
---
- |+
  \

...
```

**Actual behavior**

Strings printed in YAML consisting '\t and \v' **are not** interpreted.

**Expected behavior**

Strings printed in YAML consisting '\t' and '\v' **are** not interpreted.

**Comment by Lord-KA** [comment]
The problem is that we use YAML as default output in console. By definition it should be easy to read as an interface, but here if a module (in my case it was `tarantool.compat.dump()`) tries to output a complex string for a human to read, like `help` or some pretty dump, when it uses '\t' to e.g. align something user sees exactly escaped '\t'. I had to use multiple spaces instead and it doesn't seem like a good alternative.

**Comment by Totktonada** [comment]
I agree that #3012 would be pretty useful for a prose text (help messages and so on). However I disagree that tab escaping is a problem.

The only known use case is a formatted text within a string, like formatted help message. A tab may be rendered differently (different terminal emulators, text editors, documentation/emails/messaging apps, where the text will be pasted and so on) and it is better to always use whitespaces for formatting.

**Comment by Totktonada** [comment]
Several, but how many?

```
$ echo -e 'foo\tbar\nfoofoof\tbar'
foo	bar
foofoof	bar
```

This output looks differently in urxvt and in Vim. It also looks differently on GitHub with and without refined-github. There is no one way to show tabs.

```
$ echo -e 'foo\tbar\nfoofoofo\tbar'
foo	bar
foofoofo	bar
```

This output doesn't form columns in urxvt, but it is unclear, whether it'll do in other terminals.
